pallets/gear/src/lib.rs

```diff
@@ -742,35 +754,21 @@ pub mod pallet {
                         b"Internal error: unable to get gas limit after execution".to_vec()
                     })?;
 
+                let message_execution_context = MessageExecutionContext {
+                    actor,
+                    dispatch: queued_dispatch.into_incoming(gas_limit),
+                    origin: ProgramId::from_origin(source),
+                };
+
                 let journal = if lazy_pages_enabled {
                     core_processor::process::<LazyPagesExt, SandboxEnvironment<_>>(
-                        Some(actor),
-                        queued_dispatch.into_incoming(gas_limit),
-                        block_info,
-                        allocations_config,
-                        existential_deposit,
-                        ProgramId::from_origin(source),
-                        actor_id,
-                        u64::MAX,
-                        T::OutgoingLimit::get(),
-                        schedule.host_fn_weights.clone().into_core(),
-                        ["gr_gas_available"].into(),
-                        T::MailboxThreshold::get(),
+                        &block_config,
+                        message_execution_context,
                     )
                 } else {
                     core_processor::process::<Ext, SandboxEnvironment<_>>(
-                        Some(actor),
-                        queued_dispatch.into_incoming(gas_limit),
-                        block_info,
-                        allocations_config,
-                        existential_deposit,
-                        ProgramId::from_origin(source),
-                        actor_id,
```

Comment:
Inserting some Processor entity will also allow us to remove copy-base code, cause it already will include any generics inside it

Reply:
```rust
let processor_1 = Processor::<A, B>::new();
let processor_2 = Processor::<C, D>::new();
```

or even hide it under the hood by features, but that's not the best solution